File: frontend/src/types/realtime.ts
export interface RealtimeLive {
  trip_id: string;
  status: string | null;
  vehicle_id: string | null;
  vehicle_status: string | null;
  current_lat: number | null;
  current_lon: number | null;
  current_speed_kmph: number | null;
  alert_flag: string | null;
  last_updated: string | null;
  breadcrumb_count: number;
  latest_speed_kmph: number | null;
  latest_heading_deg: number | null;
  latest_timestamp: string | null;
}

File: backend/app/schemas/realtime.py
from datetime import datetime

from pydantic import BaseModel


class RealtimeLiveRead(BaseModel):
    trip_id: str
    status: str | None = None
    vehicle_id: str | None = None
    vehicle_status: str | None = None
    current_lat: float | None = None
    current_lon: float | None = None
    current_speed_kmph: float | None = None
    alert_flag: str | None = None
    last_updated: datetime | None = None
    breadcrumb_count: int = 0
    latest_speed_kmph: float | None = None
    latest_heading_deg: int | None = None
    latest_timestamp: datetime | None = None
